runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/operation/SdkOperationExecution.kt

```diff
@@ -191,10 +191,13 @@ private suspend fun httpTraceMiddleware(request: SdkHttpRequest, next: Handler<S
 
     var call = next.call(request)
 
+    val statusMsg = "HttpResponse: Status ${call.response.status}"
     if (logMode.isEnabled(SdkLogMode.LogResponse) || logMode.isEnabled(SdkLogMode.LogResponseWithBody)) {
         val (resp, formattedResp) = dumpResponse(call.response, logMode.isEnabled(SdkLogMode.LogResponseWithBody))
         call = call.copy(response = resp)
-        logger.debug { "HttpResponse:\n$formattedResp" }
+        logger.debug { "$statusMsg\n$formattedResp" }
```

Comment:
fix: `dumpResponse` is no longer a standalone function and now requires the status line to be written by the caller. Can we just leave `dumpResponse` alone and let the `else` branch do it's own thing:

```kotlin
if (logMode.isEnabled(...) ... ) {
    // same as before
}else {
     logger.debug { "HttpResponse: Status ${call.response.status}" }
}
```